test/unit/interval.c

```diff
@@ -96,11 +96,62 @@ test_interval_encode_decode(void)
 	is(result.adjust, DT_EXCESS, "Adjust value is right");
 }
 
+static void
+test_interval_encode_decode_limits(void)
+{
+	struct interval itv;
+	memset(&itv, 0, sizeof(itv));
+	struct interval result;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.day = -4295055470;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.day = 4295055470;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.hour = -103081331286;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.hour = 103081331286;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.min = -6184879877160;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.min = 6184879877160;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.sec = -371092792629600;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.sec = 371092792629600;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.nsec = -2147483647;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
+
+	itv.nsec = 2147483647;
+	interval_mp_recode(&itv, &result);
+	ok(is_interval_equal(&itv, &result), "Intervals are equal.");
```

Comment:
I would not use magic number and use constants defined in `limits.h` [^1][^2] instead.

[^1]: https://en.cppreference.com/w/c/types/limits
[^2]: https://learn.microsoft.com/ru-ru/cpp/c-language/cpp-integer-limits

Reply:
Updated to test on INT_MAX + 1, INT_MIN - 1, where allowed.

Reply:
Thanks!